stdlib/public/Cxx/CxxDictionary.swift

```diff
@@ -38,4 +38,20 @@ extension CxxDictionary {
       return iter.pointee.second
     }
   }
+  
+  func filter(
+    _ isIncluded: (Key, Value) throws -> Bool
+  ) rethrows -> [Key: Value] {
+    var filteredDictionary: [Key: Value] = [:]
+    let iterator = __findUnsafe(Key)
+    let endIterator = __endUnsafe()
+
+    while iterator != endIterator {
+      let pair = iterator.pointee
+      if try isIncluded(pair.first, pair.second) {
+        filteredDictionary[pair.first] = pair.second
+      }
+      iterator.successor()
```

Comment:
`.successor()` doesn't mutate the object, so this call will have no effect.

Reply:
I took a look at the usage of `rawIterator` and I saw that it's mutating with something like this one in `CxxSequence`
`self.rawIterator = self.rawIterator.successor()`
What do you think about this one?

Reply:
Yes, this looks correct to me